packages/webpack-cli/src/webpack-cli.ts

```diff
@@ -1,2544 +1,1931 @@
-          createJsonStringifyStream(stats.toJson(statsOptions as StatsOptions))
-            .on("error", handleWriteError)
-            .pipe(process.stdout)
-            .on("error", handleWriteError)
-            .on("close", () => process.stdout.write("\n"));
-        } else {
-          createJsonStringifyStream(stats.toJson(statsOptions as StatsOptions))
-            .on("error", handleWriteError)
-            .pipe(fs.createWriteStream(options.json))
-            .on("error", handleWriteError)
-            // Use stderr to logging
-            .on("close", () => {
-              process.stderr.write(
-                `[webpack-cli] ${this.colors.green(
-                  `stats are successfully stored as json to ${options.json}`,
-                )}\n`,
-              );
-            });
+        const env = isWatchCommand || options.watch
+            ? Object.assign({ WEBPACK_WATCH: true }, options.env) : Object.assign({ WEBPACK_BUNDLE: true, WEBPACK_BUILD: true }, options.env);
+        options.argv = Object.assign(Object.assign({}, options), { env });
+        if (isWatchCommand) {
+            options.watch = true;
+            options.isWatchingLikeCommand = true;
         }
-      } else {
-        const printedStats = stats.toString(
-          // TODO fix me in webpack
-          statsOptions as Exclude<WebpackOptionsNormalized["stats"], boolean>,
-        );
-
-        // Avoid extra empty line when `stats: 'none'`
-        if (printedStats) {
-          this.logger.raw(printedStats);
+        compiler = await this.createCompiler(options, callback);
+        if (!compiler) {
+            return;
+        }
+        const isWatch = (compiler) => Boolean(this.isMultipleCompiler(compiler)
+            ? compiler.compilers.some((compiler) => compiler.options.watch)
+            : compiler.options.watch);
+        if (isWatch(compiler) && this.needWatchStdin(compiler)) {
+            process.stdin.on("end", () => {
+                process.exit(0);
+            });
+            process.stdin.resume();
         }
-      }
-    };
-
-    const env =
-      isWatchCommand || options.watch
-        ? { WEBPACK_WATCH: true, ...options.env }
-        : { WEBPACK_BUNDLE: true, WEBPACK_BUILD: true, ...options.env };
-
-    options.argv = { ...options, env };
-
-    if (isWatchCommand) {
-      options.watch = true;
-      options.isWatchingLikeCommand = true;
-    }
-
-    compiler = await this.createCompiler(options as WebpackDevServerOptions, callback);
-
-    if (!compiler) {
-      return;
-    }
-
-    const isWatch = (compiler: WebpackCompiler): boolean =>
-      Boolean(
-        this.isMultipleCompiler(compiler)
-          ? compiler.compilers.some((compiler) => compiler.options.watch)
-          : compiler.options.watch,
-      );
-
-    if (isWatch(compiler) && this.needWatchStdin(compiler)) {
-      process.stdin.on("end", () => {
-        process.exit(0);
-      });
-      process.stdin.resume();
     }
-  }
 }
-
```

Comment:
Don't do these changes, we need `--dont-env` flag and apply your plugin when it is try